### ISSUE TITLE
Remove SolverInterfaceImpl::resetWrittenData?

### ISSUE THREAD

**Issue by BenjaminRodenberg** [precice/precice#1156]
Can we safely remove the function `SolverInterfaceImpl::resetWrittenData`? To me the reason why this function is needed is unclear. Additionally, there are no tests that fail, if I remove the calls to this function. Is this function really needed? If yes, a correspondingly test should be written to show this use-case. If no, it should be removed (I will create a pull request in a moment).

**Comment by fsimonis** [comment]
If I remember correctly, @DavidSCN disabled the data reset whilst developing the direct access functionality.
Is this just a left-over of these changes?

**Comment by uekerman** [comment]
IIRC, this method is a pure safeguard. Users should no longer be able to use the values in the buffer. We removed it when implementing manifold mapping (as we needed to access the values then on the user side) and re-enabled it once we removed the manifold mapping.
It is definitely bad that we do not even have a test for this.
If there is a good reason, we could remove it, but we should properly document why we did it and that we did it. It has a side effect on the user side. Maybe some users currently rely on the resetting as they sum up value while iterating over cells etc.
